owncloudData/src/test/java/com/owncloud/android/data/server/datasources/OCRemoteServerInfoDataSourceTest.kt

```diff
@@ -0,0 +1,187 @@
+
+class OCRemoteServerInfoDataSourceTest {
+    private lateinit var ocRemoteServerInfoDatasource: OCRemoteServerInfoDataSource
+
+    private val ocInfoService: OCServerInfoService = mockk()
+
+    private val OC_OWNCLOUD_VERSION = OwnCloudVersion("10.3.2")
+    private val basicAuthHeader = "basic realm=\"owncloud\", charset=\"utf-8\""
+    private val bearerHeader = "bearer realm=\"owncloud\""
+    private val redirectedLocation = "http://demo.owncloud.demo.com"
+
+    @Before
+    fun init() {
+        ocRemoteServerInfoDatasource = OCRemoteServerInfoDataSource(ocInfoService)
+    }
+
+    @Test
+    fun getAuthenticationMethodFollowRedirections() {
+        val checkPathExistenceResultFollowRedirectionMocked: RemoteOperationResult<Boolean> =
+            createRemoteOperationResultMock(
+                data = true,
+                isSuccess = true,
+                redirectedLocation = OC_ServerInfo.baseUrl
+            )
+        val checkPathExistenceResultMocked: RemoteOperationResult<Boolean> =
+            createRemoteOperationResultMock(
+                data = true,
+                isSuccess = true,
+                resultCode = OK_SSL,
+                authenticationHeader = basicAuthHeader,
+                httpCode = HTTP_UNAUTHORIZED
+            )
+
+        every {
+            ocInfoService.checkPathExistence(redirectedLocation, false)
+        } returns checkPathExistenceResultFollowRedirectionMocked
+
+        every {
+            ocInfoService.checkPathExistence(OC_ServerInfo.baseUrl, false)
+        } returns checkPathExistenceResultMocked
+
+        val authenticationMethod = ocRemoteServerInfoDatasource.getAuthenticationMethod(redirectedLocation)
+
+        assertNotNull(authenticationMethod)
+        assertEquals(AuthenticationMethod.BASIC_HTTP_AUTH, authenticationMethod)
+
+        verify { ocInfoService.checkPathExistence(OC_ServerInfo.baseUrl, false) }
+    }
+
+    @Test
+    fun getAuthenticationMethodBasic() {
+        val checkPathExistenceResultMocked: RemoteOperationResult<Boolean> =
+            createRemoteOperationResultMock(
+                data = true,
+                isSuccess = true,
+                resultCode = OK_SSL,
+                authenticationHeader = basicAuthHeader,
+                httpCode = HTTP_UNAUTHORIZED,
+                redirectedLocation = null
+            )
+
+        every {
+            ocInfoService.checkPathExistence(OC_ServerInfo.baseUrl, false)
+        } returns checkPathExistenceResultMocked
+
+        val authenticationMethod = ocRemoteServerInfoDatasource.getAuthenticationMethod(OC_ServerInfo.baseUrl)
+
+        assertNotNull(authenticationMethod)
+        assertEquals(AuthenticationMethod.BASIC_HTTP_AUTH, authenticationMethod)
+
+        verify { ocInfoService.checkPathExistence(OC_ServerInfo.baseUrl, false) }
+    }
+
+    @Test
+    fun getAuthenticationMethodBearer() {
+        val checkPathExistenceResultMocked: RemoteOperationResult<Boolean> =
```

Comment:
You could create some methods to not duplicate so much code. `getAuthenticationMethodBasic` and `getAuthenticationMethodBearer` tests are almost the same.